tests/test_api.py

```diff
@@ -125,3 +125,54 @@ def test_record_performance_file_name(self):
         perf_name = __file__.replace('.py', '.file_name.perf.yml')
         with self.record_performance(file_name=perf_name):
             caches['default'].get('foo')
+
+
+class FileNameTests(TestCase):
+
+    default_perf_name = __file__.replace('.py', '.perf.yml')
+    custom_perf_name = 'custom.perf.yml'
+    custom_path_perf_name = os.path.join(
+        os.path.dirname(__file__),
+        'performance_tests',
+        os.path.basename(__file__).replace('.py', '.perf.yml')
+    )
+
+    def setUp(self):
+        super(FileNameTests, self).setUp()
+        self.ensure_no_file()
+
+    def tearDown(self):
+        self.ensure_no_file()
+        super(FileNameTests, self).tearDown()
+
+    def ensure_no_file(self):
+        for name in (self.custom_perf_name, self.custom_path_perf_name):
+            try:
+                os.unlink(name)
+            except OSError as exc:
+                if exc.errno != errno.ENOENT:
+                    raise
+
+        try:
+            os.rmdir(os.path.dirname(self.custom_path_perf_name))
+        except OSError as exc:
+            if exc.errno != errno.ENOENT:
+                raise
```

Comment:
I didn't like the way this logic was going, getting pretty complicated and decoupled from the tests it's used in. I made a new helper `temporary_path` and used it here and in the `RecordBlankFileTests` and it allowed me to then merge all the tests that just check the behaviour of `record()` back into `RecordTests`